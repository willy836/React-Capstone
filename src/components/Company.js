import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { BsChevronRight } from 'react-icons/bs';

import '../styles/company.css';

const Company = (props) => {
  const { companySymbol, companyName, companyHeadquarter } = props;
  return (
    <div className="company-container">
      <Link to={`/company/${companySymbol}`}>
        <BsChevronRight className="chev-right" />
      </Link>
      <div className="company">
        <h5>{companySymbol}</h5>
        <h3>{companyName}</h3>
        <p>{companyHeadquarter}</p>
      </div>
    </div>
  );
};

Company.propTypes = {
  companyName: PropTypes.string,
  companyHeadquarter: PropTypes.string,
}.isRequired;

export default Company;
