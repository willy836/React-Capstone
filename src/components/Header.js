import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/header.css';

const Header = () => (
  <div>
    <nav>
      <Link to="/" className="homepage">
        Homepage
      </Link>
    </nav>
  </div>
);

export default Header;
