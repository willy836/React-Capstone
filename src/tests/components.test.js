import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import Company from '../components/Company';

describe('Test react components', () => {
  it('Header component renders correctly', () => {
    const header = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  it('Company component renders correctly', () => {
    const company = renderer
      .create(
        <Router>
          <Company />
        </Router>,
      )
      .toJSON();
    expect(company).toMatchSnapshot();
  });

  it('Companies component renders correctly', () => {
    const Companies = () => {
      <div className="companies-homepage">
        <div className="loading">
          <h2 className="loading">Loading...</h2>
        </div>
        <div className="search-companies">
          <div className="search">
            <h4 className="search-company">Search:</h4>
            <div className="input-icon">
              <input
                type="text"
                value="search"
                onChange="handleSearch"
                placeholder="Search Company"
                className="search-input"
              />
            </div>
          </div>
          <div className="display-companies">
            <h1 className="companies-h1">Companies</h1>
            <div className="companies">Companies Array data</div>
          </div>
        </div>
      </div>;
    };
    const companies = renderer
      .create(
        <Router>
          <Companies />
        </Router>,
      )
      .toJSON();
    expect(companies).toMatchSnapshot();
  });

  it('SingleCompany component renders correctly', () => {
    const SingleCompany = () => {
      <div className="about-company">
        <div className="nav">
          ChevronLeft icon
          <h2 className="company-info">Company Info</h2>
          <div className="nav-icons-right">Mic icon Gear icon</div>
        </div>
        <div className="company-details">
          <div className="company-image">
            <img src="url" alt="" className="img" />
          </div>
          <div className="company-info">
            <p className="data">Company Name</p>
            <p className="data">Company Symbol</p>
            <p className="data">Sector</p>
            <p className="data">Industry</p>
            <p className="data">Country</p>
            <p className="data">City</p>
            <p className="data">Company CEO</p>
            <p className="data">Company Phone</p>
            <p className="data">Company Address</p>
            <p className="data">Company Website</p>
            <p className="data">Company Description</p>
          </div>
        </div>
      </div>;
    };
    const singleCompany = renderer
      .create(
        <Router>
          <SingleCompany />
        </Router>,
      )
      .toJSON();
    expect(singleCompany).toMatchSnapshot();
  });
});
