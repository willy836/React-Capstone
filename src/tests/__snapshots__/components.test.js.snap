// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test react components Companies component renders correctly 1`] = `null`;

exports[`Test react components Company component renders correctly 1`] = `
<div
  className="company-container"
>
  <a
    href="/company/undefined"
    onClick={[Function]}
  >
    <svg
      className="chev-right"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
        fillRule="evenodd"
      />
    </svg>
  </a>
  <div
    className="company"
  >
    <h5 />
    <h3 />
    <p />
  </div>
</div>
`;

exports[`Test react components Header component renders correctly 1`] = `
<div>
  <nav>
    <a
      className="homepage"
      href="/"
      onClick={[Function]}
    >
      Homepage
    </a>
  </nav>
</div>
`;

exports[`Test react components SingleCompany component renders correctly 1`] = `null`;
